server/api/model/plan_area_changes.js

```diff
@@ -0,0 +1,77 @@
+const Model = require('./base_model');
+const Log = require('../lib/log');
+const Exception = require('./exception');
+const { Bookshelf, Knex } = require('../service/database');
+
+class PlanAreaChanges extends Model {
+	get rules () {
+		return {
+			id: ['required', 'integer'],
+			plan_id: ['required', 'integer'],
+			usage: ['text'],
+			measurement_unit: ['text'],
+			approved_state: ['text'],
+			change_to_approved_state: ['text'],
+            		total_in_detailed_plan: ['text'],
+            		total_in_mitaarit_plan: ['text'],
+            		remarks: ['text']
+		}
+
+	}
+
+	get tableName () {
+		return 'plan_area_changes';
+	}
+
+	canRead () {
+		return true;
+	}
+
+	getCollection () {
+		return this.collection();
+	}
+
+    	static async byPlanAndUsage (planId, usage) {
+		if (!planId) {
+			throw new Exception.BadRequest('Must provide planId');
+		}
+		if (!usage) {
+			throw new Exception.BadRequest('Must provide usage');
+		}		
+		Log.info(`checking for planId ${planId} and usage ${usage}`);
+		const result = await this.query('where', 'plan_id', '=', planId)
+        		.query('where', 'usage', '=', usage)
+			.fetchAll();
+		return result;
+	}
+
+	static async deleteByPlan ( planId) {
+		return this.query('where', 'plan_id', '=', planId)
+			.destroy({require: false})
+			.then(() => true);
+	}    
+
+	// adds multiple changes in one transcation
+	static async createPlanAreaChanges( data) {
+		try {
+			await Bookshelf.transaction(async (transaction) => {
+				for (const datum in data ){
+                    const currentChange = data[datum];
+                     await Knex.raw(`INSERT INTO plan_area_changes (plan_id, \`usage\`, measurement_unit, approved_state, change_to_approved_state, total_in_detailed_plan, total_in_mitaarit_plan, remarks) 
```

Comment:
why not to create  a new plan_area_changes model, set the property and then save? Much easier than entering values to an SQL query

Reply:
I did it at first but I got an error from checkit that it doesn't know how to check text or something like that..

Reply:
Can you please show me an example of the error you got? Maybe it's related to the rules on the different text fields?